libs/SoloLeveling/Functions/MiscOverrides.js

```diff
@@ -58,6 +58,142 @@ Misc.openChests = function (range) {
 	return true;
 };
 
+// Cast a skill on self, Unit or coords. Always use packet casting for caster skills becasue it's more stable.
+Skill.cast = function (skillId, hand, x, y, item) {
+	var casterSkills = [36, 38, 39, 44, 45, 47, 48, 49, 53, 54, 55, 56, 59, 64, 67, 84, 87, 92, 93, 101, 112, 121, 130, 137, 138, 146, 149, 154, 155, 225, 229, 230, 234, 240, 244, 245, 249, 250, 251, 256, 261, 262, 271, 276];
+
+	if (me.inTown && !this.townSkill(skillId)) {
+		return false;
+	}
+
+	if (!item && !me.getSkill(skillId, 1)) {
+		return false;
+	}
+
+	if (!this.wereFormCheck(skillId)) {
+		return false;
+	}
+
+	// Check mana cost, charged skills don't use mana
+	if (!item && this.getManaCost(skillId) > me.mp) {
+		// Maybe delay on ALL skills that we don't have enough mana for?
+		if (Config.AttackSkill.concat([42, 54]).concat(Config.LowManaSkill).indexOf(skillId) > -1) {
+			delay(300);
+		}
+
+		return false;
+	}
+
+	if (skillId === undefined) {
+		throw new Error("Skill.cast: Must supply a skill ID");
+	}
+
+	var i, n, clickType, shift;
+
+	if (hand === undefined) {
+		hand = 0;
+	}
+
+	if (x === undefined) {
+		x = me.x;
+	}
+
+	if (y === undefined) {
+		y = me.y;
+	}
+
+	if (!this.setSkill(skillId, hand, item)) {
+		return false;
+	}
+
+	if ((casterSkills.indexOf(skillId) > -1) || Config.PacketCasting > 1) {
+		if (SoloSettings.debugging.skillInfo) {
```

Comment:
Doing this, for the people who say their bot isn't attacking when its really because its using packet casting